app/controllers/application_controller.rb

```diff
@@ -9,4 +9,4 @@ def require_user
   def current_user
     User.find(session[:user_id]) if session[:user_id]
```

Comment:
I just noticed you are not caching the user instance. This should be `@current_user ||= User.find(session[:user_id]) if session[:user_id]`. This way it is very inefficient solution.